### PR TITLE
Make tooltip smarter

### PR DESCRIPTION
**Pull request by gabibguti**
Problems:
Prop `show` was just hanging around doing nothing. Also `defaultProps` were not being set properly.

Fixes:
Component shows/hide automatically on `mouseover`/`mouseout` events. And adjusts its width dynamically as well.

### CODE REVIEW

**Comment by augustolazaro** [comment]
IMHO the component should manage itself, ie, the logic for showing/hiding the tooltip must be in the component itself. Can we make a POC for that?

**Comment by gabibguti** [comment]
@augustolazaro 
Change tooltip behavior to require child component, and show tooltip when hover child component.

Link for testing: (tooltip at info icon)
https://dev--gc-wmr3721.mygocommerce.com/admin/mercadolivre/settings

File: react/components/Indicators/Tooltip/index.js [hunk 1]
```diff
@@ -3,30 +3,41 @@ import PropTypes from 'prop-types'
...
     return (
-      <div
-        className={`gc_tooltip z-5 absolute br2 g-pa4 bg-base-inverted-1 c-white ${styles.gc_tooltip} ${
-          this.props.className
-        }`}
-        style={{ width: this.props.width }}
-      >
-        {this.props.children}
-      </div>
+      <>
```
Comment (augustolazaro) [inline react/components/Indicators/Tooltip/index.js]: Ain't no need to use a fragment here

File: react/components/Indicators/Tooltip/index.js [hunk 2]
```diff
@@ -3,30 +3,41 @@ import PropTypes from 'prop-types'
...
+        </div>
+      </>
     )
   }
 }
 
 Tooltip.propTypes = {
-  show: PropTypes.bool,
   width: PropTypes.string,
+  label: PropTypes.string,
```
Comment (augustolazaro) [inline react/components/Indicators/Tooltip/index.js]: Not sure about this prop name 🤔 What about `text` instead?

File: react/components/Indicators/Tooltip/index.js [hunk 3]
```diff
@@ -3,30 +3,41 @@ import PropTypes from 'prop-types'
 import styles from './style.css'
 
 class Tooltip extends Component {
```
Comment (augustolazaro) [inline react/components/Indicators/Tooltip/index.js]: We can avoid a class and write a function component instead

File: react/components/Indicators/Tooltip/index.js [hunk 4]
```diff
@@ -3,30 +3,41 @@ import PropTypes from 'prop-types'
...
-        style={{ width: this.props.width }}
-      >
-        {this.props.children}
-      </div>
+      <>
+        <div className="dib mr2 relative">
+          <div
+            className={`gc_tooltip z-5 absolute br2 g-pa2 bg-base-inverted-1 g-f1 c-white tc
+            ${styles.gc_tooltip} ${this.props.className} ${this.state.show ? '' : 'dn'}`}
+            style={{ width: this.props.width }}
```
Comment (augustolazaro) [inline react/components/Indicators/Tooltip/index.js]: Maybe use `maxWidth` instead of `width`

**Pull request merged**